docs/ci.md

```diff
@@ -0,0 +1,213 @@
+generated binaries. For example:
+
+```
+chmod +x $GOPATH/src/k8s.io/kubernetes/_output/bin/deepcopy-gen
+```
+
+### Install KIND
+
+Kubernetes in Docker (KIND) is used to deploy Kubernetes locally where a docker
+container is created per Kubernetes node. The CI tests run on this Kubernetes
+deployment. Therefore, KIND will need to be installed locally.
+
+Installation instructions can be found at:
+https://github.com/kubernetes-sigs/kind#installation-and-usage.
+
+NOTE: The OVN-Kubernetes 
+[kind.sh](https://github.com/ovn-org/ovn-kubernetes/blob/master/contrib/kind.sh)
+and
+[kind.yaml](https://github.com/ovn-org/ovn-kubernetes/blob/master/contrib/kind.yaml)
+files provision port 11337. If firewalld is enabled, this port will need to be
+unblocked:
+
+```
+sudo firewall-cmd --permanent --add-port=11337/tcp; sudo firewall-cmd --reload
+```
+
+### Run KIND Deployment from OVN-Kubernetes
+
+To launch the KIND deployment, the `kind.sh` script must be run out of the
+OVN-Kubernetes repo. Download and build the OVN-Kubernetes repo:
+
+```
+$ cd $GOPATH/src
+$ go get github.com/ovn-org/ovn-kubernetes; cd github.com/ovn-org/ovn-kubernetes
+
+$ pushd go-controller
+$ make
+$ popd
+
+$ pushd dist/images
+$ make fedora
+$ popd
+```
+
+Launch the KIND Deployment.
+
+```
+$ pushd contrib
+$ KUBECONFIG=/home/$USER/admin.conf
+$ KIND_INSTALL_INGRESS=true ./kind.sh
+$ popd
+```
+
+This will launch a KIND deployment. The `kind.sh` script defaults the cluster
+name to `ovn`.
+
+```
+$ kubectl get nodes
+NAME                STATUS   ROLES    AGE    VERSION
+ovn-control-plane   Ready    master   3d5h   v1.16.4
+
+$ kubectl get pods --all-namespaces
+NAMESPACE            NAME                                        READY   STATUS    RESTARTS   AGE
+ingress-nginx        nginx-ingress-controller-7f74f657bd-7xcxm   1/1     Running   0          3d5h
+kube-system          coredns-5644d7b6d9-4zfkt                    1/1     Running   0          3d5h
+kube-system          coredns-5644d7b6d9-dwsgt                    1/1     Running   0          3d5h
+kube-system          etcd-ovn-control-plane                      1/1     Running   0          3d5h
+kube-system          kube-apiserver-ovn-control-plane            1/1     Running   0          3d5h
+kube-system          kube-controller-manager-ovn-control-plane   1/1     Running   0          3d5h
+kube-system          kube-scheduler-ovn-control-plane            1/1     Running   0          3d5h
+local-path-storage   local-path-provisioner-7745554f7f-kcdvj     1/1     Running   0          3d5h
+ovn-kubernetes       ovnkube-db-797db98b7c-x46rc                 2/2     Running   0          3d5h
+ovn-kubernetes       ovnkube-master-7985f79bd6-dlhnf             3/3     Running   0          3d5h
+ovn-kubernetes       ovnkube-node-jkvl9                          3/3     Running   0          3d5h
+```
+
+Once the testing is complete, to tear down the KIND deployment:
+
+```
+$ kind delete cluster --name ovn
+```
+
+### Run Tests
+
```

Comment:
There are make targets to run these tests in `test/Makefile`
You should be able to run `make shard-n` or `make control-plane` vs. having to extract that info from bash scripts

Reply:
Thanks. Done.